File: app/src/main/java/com/naosim/dddwork/domain/WorkDate.java
package com.naosim.dddwork.domain;

import java.time.LocalDate;

class WorkDate {
    private final LocalDate yearMonth;

    WorkDate(int year, int month, int day) {
        yearMonth = LocalDate.of(year, month, day);
    }

    WorkDate(WorkDate workDate) {
        this.yearMonth = workDate.yearMonth;
    }

    LocalDate getYearMonth() {
        return yearMonth;
    }
}

File: src/main/java/com/naosim/dddwork/domain/vo/WorkDate.kt
package com.naosim.dddwork.domain.vo

import java.time.LocalDate

class WorkDate(
        value: LocalDate
) : Date(value) {
    companion object {
        fun of(year: Year, month: Month, day: Day): WorkDate = WorkDate(LocalDate.of(
                year.value, month.value, day.value
        ))

        fun of(date: Date): WorkDate = WorkDate(date.value)
    }
}
